base_layer/wallet/tests/transaction_service/service.rs

```diff
@@ -908,6 +910,146 @@ fn recover_one_sided_transaction() {
+
+    let shutdown = Shutdown::new();
+    let (mut alice_ts, mut alice_oms, _alice_comms, mut alice_connectivity) = setup_transaction_service(
+        &mut runtime,
+        alice_node_identity,
+        vec![],
+        factories.clone(),
+        db_connection,
+        database_path,
+        Duration::from_secs(0),
+        shutdown.to_signal(),
+    );
+
+    let (mut bob_ts, mut bob_oms, _bob_comms, _bob_connectivity) = setup_transaction_service(
+        &mut runtime,
+        bob_node_identity.clone(),
+        vec![],
+        factories.clone(),
+        bob_connection,
+        database_path2,
+        Duration::from_secs(0),
+        shutdown.to_signal(),
+    );
+
+    let mut alice_event_stream = alice_ts.get_event_stream();
+
+    alice_connectivity.set_base_node(base_node_identity.to_peer());
+
+    let initial_wallet_value = 2500.into();
+    let (_utxo, uo1) = make_input(&mut OsRng, initial_wallet_value, &factories.commitment);
+    let mut alice_oms_clone = alice_oms.clone();
+    runtime.block_on(async move { alice_oms_clone.add_output(uo1).await.unwrap() });
+
+    let message = "".to_string();
+    let value = 1000.into();
+    let mut alice_ts_clone = alice_ts.clone();
+    let (tx_id, pre_image, output) = runtime.block_on(async move {
+        alice_ts_clone
+            .send_sha_atomic_swap_transaction(
+                bob_node_identity.public_key().clone(),
+                value,
+                20.into(),
+                message.clone(),
+            )
+            .await
+            .expect("Alice sending HTLC transaction")
+    });
+
+    runtime.block_on(async move {
+        let completed_tx = alice_ts
+            .get_completed_transaction(tx_id)
+            .await
+            .expect("Could not find completed HTLC tx");
+
+        let fees = completed_tx.fee;
+
+        assert_eq!(
+            alice_oms.get_balance().await.unwrap().pending_incoming_balance,
+            initial_wallet_value - value - fees
+        );
+    });
+
+    runtime.block_on(async {
+        let delay = sleep(Duration::from_secs(30));
+        tokio::pin!(delay);
+        let mut found = false;
+        loop {
+            tokio::select! {
+                event = alice_event_stream.recv() => {
+                    if let TransactionEvent::TransactionCompletedImmediately(id) = &*event.unwrap() {
+                        if id == &tx_id {
+                            found = true;
+                            break;
+                        }
+                    }
+                },
+                () = &mut delay => {
+                    break;
+                },
+            }
+        }
+        assert!(found, "'TransactionCompletedImmediately(_)' event not found");
+    });
+
+    runtime.block_on(async move {
+        let (tx_id_htlc, htlc_fee, htlc_amount, tx) = bob_oms
+            .create_claim_sha_atomic_swap_transaction_with_output(output, pre_image, 20.into())
```

Comment:
So I see this is using a method that is used nowhere else? The `command.rs` file doesn't use this version. I am guessing its so that you don't have to handle the base node UTXO request? I would really prefer you actually test that part by using a base node RPC mock. You would use `setup_transaction_service_no_comms` which provides you with a RPC mock